server.go

```diff
@@ -118,58 +121,68 @@ func mutateHandler(w http.ResponseWriter, r *http.Request) {
 		}
 
 		// Define the env var
-		e := corev1.EnvVar{
-			Name:  "GOOGLE_APPLICATION_CREDENTIALS",
-			Value: "/google-app-creds.json",
+		if needsCreds {
+			e := corev1.EnvVar{
+				Name:  "GOOGLE_APPLICATION_CREDENTIALS",
+				Value: "/google-app-creds.json",
+			}
+			envVars = append(envVars, e)
 		}
-		envVars := []corev1.EnvVar{e}
 
 		// If GOOGLE_CLOUD_PROJECT is set in the VM, set it for all GCP apps.
 		if _, err := os.Stat("/var/lib/minikube/google_cloud_project"); err == nil {
 			project, err := ioutil.ReadFile("/var/lib/minikube/google_cloud_project")
 			if err == nil {
 				// Set the project name for every variant of the project env var
 				for _, a := range projectAliases {
-					envVars = append(envVars, corev1.EnvVar{
-						Name:  a,
-						Value: string(project),
-					})
+					if needsEnvVar(pod.Spec.Containers[0], a) {
+						envVars = append(envVars, corev1.EnvVar{
+							Name:  a,
+							Value: string(project),
+						})
+					}
 				}
 			}
 		}
 
-		patch = append(patch, patchOperation{
-			Op:    "add",
-			Path:  "/spec/volumes",
-			Value: append(pod.Spec.Volumes, v),
-		})
-
-		for i, c := range pod.Spec.Containers {
-			if len(c.VolumeMounts) == 0 {
-				patch = append(patch, patchOperation{
-					Op:    "add",
-					Path:  fmt.Sprintf("/spec/containers/%d/volumeMounts", i),
-					Value: []corev1.VolumeMount{mount},
-				})
-			} else {
-				patch = append(patch, patchOperation{
-					Op:    "add",
-					Path:  fmt.Sprintf("/spec/containers/%d/volumeMounts", i),
-					Value: append(c.VolumeMounts, mount),
-				})
-			}
-			if len(c.Env) == 0 {
-				patch = append(patch, patchOperation{
-					Op:    "add",
-					Path:  fmt.Sprintf("/spec/containers/%d/env", i),
-					Value: envVars,
-				})
-			} else {
-				patch = append(patch, patchOperation{
-					Op:    "add",
-					Path:  fmt.Sprintf("/spec/containers/%d/env", i),
-					Value: append(c.Env, envVars...),
-				})
+		if needsCreds {
+			patch = append(patch, patchOperation{
+				Op:    "add",
+				Path:  "/spec/volumes",
+				Value: append(pod.Spec.Volumes, v),
+			})
```

Comment:
Could you combine this `if` with the identical one on L124?

Reply:
good call, done